src/abstract-interpretation/domain.ts

```diff
@@ -166,4 +184,82 @@ export function subtractDomains(domain1: Domain, domain2: Domain): Domain {
 		}
 	}
 	return Domain.fromIntervals(intervals)
+}
+
+export const enum NarrowKind {
+	Equal = 1,
+	Smaller = 2,
+	Greater = 4
+}
+
+interface IntervalOverlap {
+	smaller:      Interval | undefined,
+	intersection: Interval | undefined,
+	larger:       Interval | undefined
+}
+
+function flipInclusiveness(intervalBound: IntervalBound): IntervalBound {
+	return {value: intervalBound.value, inclusive: !intervalBound.inclusive}
+
+}
+
+export function overlapIntervals(interval1: Interval, interval2: Interval): IntervalOverlap {
```

Comment:
warum das und `doIntervalsOverlap`?

Reply:
`overlapIntervals` currently uses `doIntervalsOverlap`, but this could definitly be refactored, then `doIntervalsOverlap` would be no longer needed (or maybe just as a small wrapper for `overlapIntervals`).